src/kernel/utils.c

```diff
@@ -0,0 +1,60 @@
+#include "utils.h"
+
+void copy_memory(char* source, char* dest, int num_bytes) {
+	for (int i = 0; i < num_bytes; i++) {
+		*(dest + i) = *(source + i);
+	}
+}
+
+void itoa(int num, char* str) {
+	int i = 0;
```

Comment:
Check all pointers in this file

Reply:
done, but to fix it more i'll have to make all functions return an error code as well in the future since the caller has no idea the function failed

Reply:
added more error handling